### PR TITLE
Merge sync node to develop

### PR DESCRIPTION
**Pull request by dmytrotkk**
This PR merges sync-node branch to the develop.

Manual merge, minor additional changes to fix build and sChain mountpoint.
No additional tests needed.
No performance changes.

Fixes https://github.com/skalenetwork/internal-support/issues/907

### CODE REVIEW

File: node_cli/configs/__init__.py [hunk 1]
```diff
@@ -32,9 +32,11 @@
 G_CONF_HOME = os.getenv('TEST_HOME_DIR') or GLOBAL_CONFIG['home_dir']
 
 SKALE_STATE_DIR = '/var/lib/skale'
+SCHAIN_STATE_DIR = os.path.join('/var/lib/skale', 'schains')
```
Comment (badrogger) [inline node_cli/configs/__init__.py]: duplicate of SCHAINS_MNT_DIR_SYNC

File: tests/conftest.py [hunk 2]
```diff
@@ -193,6 +197,19 @@ def mocked_g_config():
...
+@pytest.fixture()
+def clean_node_options():
+    pathlib.Path(NODE_OPTIONS_FILEPATH).unlink(missing_ok=True)
+    try:
+        yield
+    finally:
+        pathlib.Path(NODE_OPTIONS_FILEPATH).unlink(missing_ok=True)
+
+
+def clean_node_data():
```
Comment (badrogger) [inline tests/conftest.py]: unused

File: node_cli/utils/docker_utils.py [hunk 3]
```diff
@@ -220,7 +224,15 @@ def get_up_compose_cmd(services):
...
-def compose_up(env):
+def get_up_compose_sync_cmd():
+    return ('docker-compose', '-f', SYNC_COMPOSE_PATH, 'up', '-d')
+
+
+def get_compose_path(sync_node: bool) -> str:
+    return SYNC_COMPOSE_PATH if sync_node else COMPOSE_PATH
+
+
+def compose_up(env, sync_node=False):
```
Comment (badrogger) [inline node_cli/utils/docker_utils.py:235]: Unused `sync_node=False` here. May be it's better to merge it with `compose_up_sync` instead.

**Review by badrogger — CHANGES_REQUESTED** [review]

File: node_cli/configs/__init__.py (hunk 1 above)
Comment (dmytrotkk) [reply node_cli/configs/__init__.py]: removed `SCHAIN_STATE_DIR`

File: tests/conftest.py (hunk 2 above)
Comment (dmytrotkk) [reply tests/conftest.py]: removed

File: node_cli/utils/docker_utils.py (hunk 3 above)
Comment (dmytrotkk) [reply node_cli/utils/docker_utils.py:235]: merged `compose_up_sync` info `compose_up`

**Review by badrogger — APPROVED** [review]

**Pull request merged**